source/common/http/http2/codec_impl.cc

```diff
@@ -408,7 +492,32 @@ void ConnectionImpl::ClientStreamImpl::decodeHeaders() {
   }
 }
 
+bool ConnectionImpl::StreamImpl::maybeDeferDecodeTrailers() {
+  // Buffer trailers if we're deferring processing and not flushing all data
+  // through and either
+  // 1) Buffers are overrun
+  // 2) There's buffered body which should get processed before these trailers
+  //    to avoid losing data.
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.defer_processing_backedup_streams") &&
```

Comment:
So if this flag was true, we buffered some body, we flip the flag false, and get trailers, what happens?
Please run through this PR and make sure it's safe to flip the flag from true -> false or false -> true at any point, or else latch it on stream creation for simplicity :-)